plugins/meta/route-override/route-override.go

```diff
@@ -0,0 +1,420 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is a "meta-plugin". It reads in its own netconf, it does not create
+// any network interface but just changes route information given from
+// previous cni plugins
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+	"github.com/containernetworking/plugins/pkg/ns"
+
+	"github.com/vishvananda/netlink"
+)
+
+// RouteOverrideConfig represents the network route-override configuration
+type RouteOverrideConfig struct {
+	types.NetConf
+
+	FlushRoutes  bool           `json:"flushExisting,omitempty"`
+	FlushGateway bool           `json:"flushDefaultGateway,omitempty"`
+	DelRoutes    []*types.Route `json:"del"`
+	AddRoutes    []*types.Route `json:"add"`
+	SkipCheck    bool           `json:"skipcheck,omitempty"`
+
+	Args *struct {
+		A *IPAMArgs `json:"cni"`
+	} `json:"args"`
+}
+
+// IPAMArgs represents CNI argument conventions for the plugin
+type IPAMArgs struct {
+	FlushRoutes  *bool          `json:"flushExisting,omitempty"`
+	FlushGateway *bool          `json:"flushDefaultGateway,omitempty"`
+	DelRoutes    []*types.Route `json:"del,omitempty"`
+	AddRoutes    []*types.Route `json:"add,omitempty"`
+	SkipCheck    *bool          `json:"skipcheck,omitempty"`
+}
+
+func parseConf(data []byte, envArgs string) (*RouteOverrideConfig, error) {
+	conf := RouteOverrideConfig{FlushRoutes: false}
+
+	if err := json.Unmarshal(data, &conf); err != nil {
+		return nil, fmt.Errorf("failed to load netconf: %v", err)
+	}
+
+	// override values by args
+	if conf.Args != nil {
+		if conf.Args.A.FlushRoutes != nil {
+			conf.FlushRoutes = *conf.Args.A.FlushRoutes
+		}
+
+		if conf.Args.A.FlushGateway != nil {
+			conf.FlushGateway = *conf.Args.A.FlushGateway
+		}
+
+		if conf.Args.A.DelRoutes != nil {
+			conf.DelRoutes = conf.Args.A.DelRoutes
+		}
+
+		if conf.Args.A.AddRoutes != nil {
+			conf.AddRoutes = conf.Args.A.AddRoutes
+		}
+
+		if conf.Args.A.SkipCheck != nil {
+			conf.SkipCheck = *conf.Args.A.SkipCheck
+		}
+
+	}
+
+	// Parse previous result
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("could not serialize prevResult: %v", err)
+		}
+
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not parse prevResult: %v", err)
+		}
+
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+
+	return &conf, nil
+}
+
+func deleteAllRoutes(res *current.Result) error {
+	var err error
```

Comment:
I cannot remove it because 

- golang requires type declaration (as you may know about it, of course) (1)
- using short variable declarations, :=, we may remove it in some case, but this case is not because of scope of variables. (2)

(1)
```
[tohayash@tohayash-srv plugins-s1061123]$ ./build_linux.sh
Building plugins
  bandwidth
  firewall
  flannel
  portmap
  route-override
# github.com/containernetworking/plugins/plugins/meta/route-override
plugins/meta/route-override/route-override.go:124:8: undefined: err
plugins/meta/route-override/route-override.go:125:11: undefined: err
plugins/meta/route-override/route-override.go:126:16: undefined: err
plugins/meta/route-override/route-override.go:130:7: undefined: err
plugins/meta/route-override/route-override.go:131:10: undefined: err
plugins/meta/route-override/route-override.go:132:15: undefined: err
plugins/meta/route-override/route-override.go:140:9: undefined: err
```

(2)
```
# github.com/containernetworking/plugins/plugins/meta/route-override
plugins/meta/route-override/route-override.go:140:9: undefined: err
```